Paths with forward slashes don't work in some cases on Windows and Linux

I reported this issue in https://github.com/editorconfig-checker/editorconfig-checker/issues/281 first, but then figured out it's in fact a problem in `editorconfig-core-go` itself that is used by [editorconfig-checker](https://github.com/editorconfig-checker/editorconfig-checker) under the hood.

Please check out https://github.com/editorconfig-checker/editorconfig-checker/issues/281 - I'll repeat some of it here, but not with all the details.

Summary: even though the [EditorConfig spec](https://spec.editorconfig.org/#file-format) declares that forward slashes `/` should always be used as path separators in section headers even on Windows, in reality forward slashes `/` don't work on Windows and only backslashes `\\` do when using `editorconfig-core-go` (contrary to the linked EditorConfig spec).

An example is given in https://github.com/editorconfig-checker/editorconfig-checker/issues/281#issue-1864000959:

**doesn't work** on Windows

```editorconfig
[bin/ksdump]
indent_size = 2
```

**works** on Windows

```editorconfig
[bin\\ksdump]
indent_size = 2
```

Another related problem is that the presence of glob with a forward slash like `bin/ksdump` on Windows actually neutralizes the entire `.editorconfig` section using brace syntax (`{s1,s2,s3}`) where `bin/ksdump` appears, as demonstrated in https://github.com/editorconfig-checker/editorconfig-checker/issues/281#issuecomment-1690645392. This can be observed in slightly changed `editorconfig-core-go` tests too - just change

https://github.com/editorconfig/editorconfig-core-go/blob/0bb51f18a1573bdc43d0e49c2365fc397765b75a/testdata/root/.editorconfig#L1-L5

as follows:

```diff
 root = true
 
-[*.go]
+[{src/a.ini,*.go}]
 indent_style = space
 insert_final_newline = true
```

and run `go test -v ./...` on Windows:

```console
pp@DESKTOP-89OPGF3 MINGW64 /c/temp/editorconfig-core-go (master)
$ go test -v ./...
...

--- FAIL: TestPublicTestDefinitionForFilename (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0xb2be66]

goroutine 13 [running]:
testing.tRunner.func1.2({0xb529e0, 0xd1d590})
	C:/Program Files/Go/src/testing/testing.go:1526 +0x24e
testing.tRunner.func1()
	C:/Program Files/Go/src/testing/testing.go:1529 +0x39f
panic({0xb529e0, 0xd1d590})
	C:/Program Files/Go/src/runtime/panic.go:884 +0x213
github.com/editorconfig/editorconfig-core-go/v2.TestPublicTestDefinitionForFilename(0x0?)
	C:/temp/editorconfig-core-go/editorconfig_test.go:160 +0xc6
testing.tRunner(0xc000192680, 0xb99210)
	C:/Program Files/Go/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
	C:/Program Files/Go/src/testing/testing.go:1629 +0x3ea
FAIL	github.com/editorconfig/editorconfig-core-go/v2	0.297s
FAIL
```

If you change `src/a.ini` to `src\\a.ini`, it works (and BTW, I checked that not only the `*.go` glob works, but the `src\\a.ini` is applied too):

```diff
 root = true
 
-[{src/a.ini,*.go}]
+[{src\\a.ini,*.go}]
 indent_style = space
 insert_final_newline = true
```

```console
$ go test -v ./...
...

--- PASS: TestPublicTestDefinitionForFilename (0.00s)
```

---

Weirdly enough, including `src/a.ini` as one of the globs in brace syntax neutralizes the section even on Linux:

```editorconfig
root = true
 
[{src/a.ini,*.go}]
indent_style = space
insert_final_newline = true
```

This is the output from a run in WSL 2:

```console
pp@DESKTOP-89OPGF3:/mnt/c/temp/editorconfig-core-go$ cat testdata/root/.editorconfig
root = true

[{src/a.ini,*.go}]
indent_style = space
insert_final_newline = true
pp@DESKTOP-89OPGF3:/mnt/c/temp/editorconfig-core-go$ make test-go
go test -v ./...
...

--- FAIL: TestPublicTestDefinitionForFilename (0.01s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5d4d26]

goroutine 41 [running]:
testing.tRunner.func1.2({0x5fa5e0, 0x7b0350})
        /usr/local/go/src/testing/testing.go:1526 +0x24e
testing.tRunner.func1()
        /usr/local/go/src/testing/testing.go:1529 +0x39f
panic({0x5fa5e0, 0x7b0350})
        /usr/local/go/src/runtime/panic.go:884 +0x213
github.com/editorconfig/editorconfig-core-go/v2.TestPublicTestDefinitionForFilename(0xc0001196c0?)
        /mnt/c/temp/editorconfig-core-go/editorconfig_test.go:160 +0xc6
testing.tRunner(0xc000119860, 0x63d7a8)
        /usr/local/go/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:1629 +0x3ea
FAIL    github.com/editorconfig/editorconfig-core-go/v2 0.029s
FAIL
make: *** [Makefile:17: test-go] Error 1
```

On the other hand, when splitting the brace syntax into two separate sections, both sections work on Linux (whereas on Windows, a path with forward slashes like `src/a.ini` wouldn't work no matter what):

```editorconfig
root = true
 
[src/a.ini]
indent_style = space
insert_final_newline = true

[*.go]
indent_style = space
insert_final_newline = true
```

I haven't dived into the code yet, so the few examples I've presented are all I have for now - hopefully you'll be able to generalize the problem based on them.

--- a/editorconfig.go
+++ b/editorconfig.go
@@ -6,7 +6,6 @@ import (
 	"fmt"
 	"io"
 	"os"
-	"runtime"
 	"strings"
 
 	"gopkg.in/ini.v1"
@@ -122,11 +121,7 @@ func (e *Editorconfig) GetDefinitionForFilename(name string) (*Definition, error
 		}
 
 		if !strings.HasPrefix(name, "/") {
-			if runtime.GOOS != "windows" {
-				name = "/" + name
-			} else {
-				name = "\\" + name
-			}
+			name = "/" + name
 		}
 
 		ok, err := e.FnmatchCase(selector, name)

--- a/fnmatch.go
+++ b/fnmatch.go
@@ -3,7 +3,6 @@ package editorconfig
 import (
 	"fmt"
 	"regexp"
-	"runtime"
 	"strconv"
 	"strings"
 )
@@ -33,7 +32,7 @@ func FnmatchCase(pattern, name string) (bool, error) {
 	return r.MatchString(name), nil
 }
 
-func translate(pattern string) string { //nolint:funlen,gocognit,gocyclo,cyclop,maintidx
+func translate(pattern string) string { //nolint:funlen,gocognit,gocyclo,cyclop
 	index := 0
 	pat := []rune(pattern)
 	length := len(pat)
@@ -51,10 +50,6 @@ func translate(pattern string) string { //nolint:funlen,gocognit,gocyclo,cyclop,
 	doubleRight := len(findDoubleRightBrackets.FindAllString(pattern, -1))
 	matchesBraces := left+doubleLeft == right+doubleRight
 	pathSeparator := "/"
-
-	if runtime.GOOS == "windows" {
-		pathSeparator = regexp.QuoteMeta("\\")
-	}
 
 	for index < length {
 		r := pat[index]

